datamapper/repo.py

```diff
@@ -117,35 +116,46 @@ def __init__(self, database: Database):
         sql = func.count().select().select_from(sql)
         return await self.database.fetch_val(sql)
 
-    async def insert(self, model: Type[Model], **values: Any) -> Model:
+    async def insert(
+        self, model_or_changeset: Union[Model, Changeset[Model]], **values: Any
+    ) -> Union[Model, Changeset]:
         """
         Insert a record into the database.
 
         Examples::
 
             await repo.insert(User, name="Fred")
+            await repo.insert(changeset)
         """
-        sql_values = _collect_sql_values(model, values)
+        changeset = cast_changeset(model_or_changeset, values)
+        if not changeset.is_valid:
+            return changeset
+
+        model = changeset.data
+        sql_values = _collect_sql_values(model, changeset.changes)
```

Comment:
I think `_collect_sql_values` can be removed now?

I don't think this is necessary anymore because of `put_assoc`.